java/src/gov/usgs/consumerclient/ConsumerClient.java

```diff
@@ -407,11 +439,26 @@ public static boolean writeMessagesToDisk(Integer numToWrite) {
 			Long timeNow = System.currentTimeMillis();
 
 			// build filename from desired output directory, time, optional
-			// name,
-			// and extension
+			// name, and extension
 			String outFileName = outputDirectory + "/" + timeNow.toString()
 					+ fileName + "." + fileExtension;
 
+			// did we just use this file name?
+			if (outFileName == lastFileName) {
```

Comment:
Seems like this is gonna alternate between 
my_file_1234567890.xml
and
my_file_1234567890_1.xml

So if there are 3 files at the same time, then you're only gonna get two of them, because you modify outFileName with the _1 extension, which changes your comparison farther up.

Reply:
Seems like the easiest way to fix the logic would be to always include the dupFileName value in the filename.
Or...
Always put a sleep(1) in between files and then include the milliseconds in the filename, so that you never need the dup_filename extension and will never have duplicate filenames(assuming that timeNow or similar has enough granularity.)

Reply:
Milliseconds have aways been in the file name, that's why I had a hard time believing this file overwriting was happening at first